python/ray/util/client/server/proxier.py

```diff
@@ -242,9 +240,20 @@ def _create_runtime_env(
         retries = 0
         max_retries = 5
         wait_time_s = 0.5
+        last_exception = None
         while retries <= max_retries:
             try:
-                r = self._runtime_env_stub.GetOrCreateRuntimeEnv(create_env_request)
+                url = urllib.parse.urljoin(
+                    self._runtime_env_agent_address, "/get_or_create_runtime_env"
+                )
+                data = create_env_request.SerializeToString()
+                req = urllib.request.Request(url, data=data, method="POST")
+                req.add_header("Content-Type", "application/octet-stream")
+                response = urllib.request.urlopen(req, timeout=None)
+                response_data = response.read()
+                r = runtime_env_agent_pb2.GetOrCreateRuntimeEnvReply()
+                r.ParseFromString(response_data)
```

Comment:
why are we sending serialized protobuf messages via HTTP? very odd pattern... why not just use grpc?

Reply:
this whole PR is about removing the existing grpc server in python for runtime env agent, because we want to get rid of the grpc python dependency.